core/templates/pages/learner-group-pages/view-group/view-learner-group-page.component.spec.ts

```diff
@@ -0,0 +1,298 @@
+        },
+        {
+          provide: TranslateService,
+          useClass: MockTranslateService
+        }
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    learnerGroupBackendApiService = TestBed.inject(
+      LearnerGroupBackendApiService);
+    translateService = TestBed.inject(TranslateService);
+    pageTitleService = TestBed.inject(PageTitleService);
+    contextService = TestBed.inject(ContextService);
+    ngbModal = TestBed.inject(NgbModal);
+    loaderService = TestBed.inject(LoaderService);
+    userService = TestBed.inject(UserService);
+    learnerGroupSyllabusBackendApiService = TestBed.inject(
+      LearnerGroupSyllabusBackendApiService);
+    fixture = TestBed.createComponent(ViewLearnerGroupPageComponent);
+    component = fixture.componentInstance;
+
+    spyOn(contextService, 'getLearnerGroupId').and.returnValue('groupId');
+
+    fixture.detectChanges();
+  });
+
+  it('should initialize', fakeAsync(() => {
+    spyOn(learnerGroupBackendApiService, 'fetchLearnerGroupInfoAsync')
+      .and.returnValue(Promise.resolve(learnerGroup));
+    spyOn(
+      learnerGroupBackendApiService,
+      'fetchProgressSharingPermissionOfLearnerAsync'
+    ).and.returnValue(Promise.resolve(true));
+    spyOn(
+      learnerGroupSyllabusBackendApiService,
+      'fetchLearnerSpecificProgressInAssignedSyllabus'
+    ).and.returnValue(Promise.resolve(sampleLearnerGroupUserProg));
+    spyOn(userService, 'getUserInfoAsync').and.returnValue(
+      Promise.resolve(userInfo));
+    spyOn(component, 'subscribeToOnLangChange');
+    spyOn(translateService.onLangChange, 'subscribe');
+
+    component.ngOnInit();
+    tick();
+
+    expect(component.subscribeToOnLangChange).toHaveBeenCalled();
+    expect(component.activeTab).toEqual(
+      LearnerGroupPagesConstants.VIEW_LEARNER_GROUP_TABS.OVERVIEW);
+    expect(component.learnerGroup).toEqual(learnerGroup);
+    expect(component.getLearnersCount()).toBe(1);
+    expect(component.username).toBe('username1');
+    expect(component.progressSharingPermission).toBe(true);
+  }));
+
+  it('should call set page title whenever the language is changed',
+    fakeAsync(() => {
+      spyOn(learnerGroupBackendApiService, 'fetchLearnerGroupInfoAsync')
+        .and.returnValue(Promise.resolve(learnerGroup));
+      spyOn(
+        learnerGroupBackendApiService,
+        'fetchProgressSharingPermissionOfLearnerAsync'
+      ).and.returnValue(Promise.resolve(true));
+      spyOn(
+        learnerGroupSyllabusBackendApiService,
+        'fetchLearnerSpecificProgressInAssignedSyllabus'
+      ).and.returnValue(Promise.resolve(sampleLearnerGroupUserProg));
+      spyOn(userService, 'getUserInfoAsync').and.returnValue(
+        Promise.resolve(userInfo));
+
+      component.ngOnInit();
+      tick();
+      spyOn(component, 'setPageTitle');
+
+      translateService.onLangChange.emit();
+
+      expect(component.setPageTitle).toHaveBeenCalled();
```

Comment:
Can you test if `setDocumentTitle` is called? It would be ideal for testing if the language changes

Reply:
@gp201 I added this test and the `setDocumentTitle` is not being called, I can't figure out why that might be happening, can you please take a look once. Thanks!